### PR TITLE
Fix `ActiveRecord` check on `Confirmable`

### PR DESCRIPTION
**Pull request by tegon**
As pointed out by @dark-panda in #4302, the condition for an `ActiveRecord` model is wrong inside the `Confirmable` initialization block.

https://github.com/plataformatec/devise/pull/4302#issuecomment-355103489

### CODE REVIEW

**Comment by rafaelfranca** [comment]
It makes sense, but since no test were failing we are missing some tests.

**Comment by tegon** [comment]
@rafaelfranca I agree, but I'm struggling to figure out which kind of test would make sense here 🤔 
We have an [integration test](https://github.com/plataformatec/devise/blob/master/test/integration/confirmable_test.rb) that covers the desired behavior - for example, a confirmation email should be sent after the user is registered. This test doesn't fail because even though we fall on the `else` condition, we still define a callback to send the email - `after_create` instead of `after_commit` - but the email is sent either way.
Do you think we need to test ActiveRecord's internal methods to check what callbacks are being defined?

**Comment by rafaelfranca** [comment]
oh, I see. Right now the only way to test this is to check the callbacks yes.

**Comment by tegon** [comment]
Got it, I've added a test.

File: lib/devise/models/confirmable.rb [hunk 1]
```diff
@@ -48,7 +48,7 @@ module Confirmable
       included do
         before_create :generate_confirmation_token, if: :confirmation_required?
         after_create :skip_reconfirmation_in_callback!, if: :send_confirmation_notification?
-        if defined?(ActiveRecord) && self.is_a?(ActiveRecord::Base) # ActiveRecord
+        if defined?(ActiveRecord) && self < ActiveRecord::Base # ActiveRecord
```
Comment (feliperenan) [inline lib/devise/models/confirmable.rb]: I think the comment `#ActiveRecord` isn't necessary. What do you think to remove it?

**Pull request merged**